edb_mustgather.sh

```diff
@@ -0,0 +1,112 @@
+mkdir ${PO}
+mkdir ${PO}/${PG_OP}
+mkdir ${CLUSTER}
+mkdir ${CLUSTER}/${EDB_CLUSTER_NAME}
+mkdir ${CLUSTER_PODS}
+mkdir ${CLUSTER_BACKUPS}
+mkdir ${CLUSTER_BACKUPS}/${EDB_CLUSTER_NAME}
+
+function gatherEdbOperatorData() {
+    kubectl describe pod ${PG_OP} -n ${EDB_OP_NAMESPACE} > ${PO}/${PG_OP}/describe.txt
+    kubectl logs ${PG_OP} -n ${EDB_OP_NAMESPACE} > ${PO}/${PG_OP}/logs.txt
+}
+
+function gatherClusterData() {
+    kubectl cnp status ${EDB_CLUSTER_NAME} -n ${EDB_CLUSTER_NAMESPACE} > ${CLUSTER}/${EDB_CLUSTER_NAME}/status.txt
+    kubectl cnp status ${EDB_CLUSTER_NAME} --verbose -n ${EDB_CLUSTER_NAMESPACE} > ${CLUSTER}/${EDB_CLUSTER_NAME}/status-verbose.txt
+    kubectl get cluster -n ${EDB_CLUSTER_NAMESPACE} > ${CLUSTER}/${EDB_CLUSTER_NAME}/info.txt
+    kubectl get cluster -o yaml -n ${EDB_CLUSTER_NAMESPACE} > ${CLUSTER}/${EDB_CLUSTER_NAME}/cluster.yaml
+    kubectl describe cluster -n ${EDB_CLUSTER_NAMESPACE} > ${CLUSTER}/${EDB_CLUSTER_NAME}/describe.txt
+
+}
+
+function gatherEDBPodData() {
+    kubectl get pod -l k8s.enterprisedb.io/cluster=${EDB_CLUSTER_NAME} -L role -n ${EDB_CLUSTER_NAMESPACE} > ${CLUSTER_PODS}/pods.txt
+    for pod in ${EDB_POD_NAMES}; do
+        mkdir ${CLUSTER_PODS}/${pod} 
+        kubectl get po ${pod} -o yaml -n ${EDB_CLUSTER_NAMESPACE} > ${CLUSTER_PODS}/${pod}/pod.yaml
+        kubectl describe pod ${pod} -n ${EDB_CLUSTER_NAMESPACE} > ${CLUSTER_PODS}/${pod}/describe.txt
+        kubectl logs ${pod} -n ${EDB_CLUSTER_NAMESPACE} > ${CLUSTER_PODS}/${pod}/logs.txt
+        kubectl logs ${pod} -n ${EDB_CLUSTER_NAMESPACE} --previous 2>/dev/null > ${CLUSTER_PODS}/${pod}/previous-logs.txt
+        kubectl logs ${pod} -n ${EDB_CLUSTER_NAMESPACE} | jq -r '.record | select(.error_severity == "FATAL")' > ${CLUSTER_PODS}/${pod}/logs-fatal.txt
+    done
+}
+
+function gatherEDBBackupData() {
+    kubectl get backups -n ${EDB_CLUSTER_NAMESPACE} > ${CLUSTER_BACKUPS}/${EDB_CLUSTER_NAME}/backups.txt
+    for backup in ${EDB_BACKUP_NAMES}; do
+        mkdir ${CLUSTER_BACKUPS}/${EDB_CLUSTER_NAME}/${backup} 
+        kubectl get backups ${backup} -o yaml -n ${EDB_CLUSTER_NAMESPACE} > ${CLUSTER_BACKUPS}/${EDB_CLUSTER_NAME}/${backup}/backup.yaml
+        kubectl describe backups ${backup} -n ${EDB_CLUSTER_NAMESPACE} > ${CLUSTER_BACKUPS}/${EDB_CLUSTER_NAME}/${backup}/describe.txt
+    done
+}
+
+
+
+if [[ -z "$PG_OP" ]]; then
+    echo "failed to find the edb operator in the ${EDB_OP_NAMESPACE} namespace"
+    exit 1
+else
+   gatherEdbOperatorData
+fi
+
+if [[ -z "$EDB_CLUSTER_NAME" ]]; then
+    echo "failed to find the edb cluster in the ${EDB_CLUSTER_NAMESPACE} namespace"
+    exit 1
+else
+   gatherClusterData
+fi
+
+if [[ -z "$EDB_POD_NAMES" ]]; then
+    echo "failed to find the edb cluster pods in the ${EDB_CLUSTER_NAMESPACE} namespace"
+    exit 1
+else
+   gatherEDBPodData
+fi
+
+if [[ -z "$EDB_BACKUP_NAMES" ]]; then
+    echo "failed to find the edb cluster backups in the ${EDB_CLUSTER_NAMESPACE} namespace"
```

Comment:
There is no guarentee that backups will always exist. So we should log and silently fail instead of exit 1. 

Customers might get confused.